src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/X509Certificates/OpenSslX509ChainEventSource.cs

```diff
@@ -0,0 +1,749 @@
+        private const int EventId_DownloadTimeExceeded = 13;
+        private const int EventId_InvalidDownloadedCertificate = 14;
+        private const int EventId_InvalidDownloadedCrl = 15;
+        private const int EventId_InvalidDownloadedOcsp = 16;
+        private const int EventId_DownloadCompleteStatusCode = 17;
+        private const int EventId_DownloadRedirected = 18;
+        private const int EventId_DownloadRedirectsExceeded = 19;
+        private const int EventId_DownloadRedirectNotFollowed = 20;
+        private const int EventId_UntrustedChainWithRevocation = 21;
+        private const int EventId_NonHttpCdpEntry = 22;
+        private const int EventId_NoCdpFound = 23;
+        private const int EventId_NoMatchingAiaEntry = 24;
+        private const int EventId_NoMatchingCdpEntry = 25;
+        private const int EventId_CrlCacheCheckStart = 26;
+        private const int EventId_CrlCacheCheckStop = 27;
+        private const int EventId_CrlCacheOpenError = 28;
+        private const int EventId_CrlCacheDecodeError = 29;
+        private const int EventId_CrlCacheExpired = 30;
+        private const int EventId_CrlCacheFileBasedExpiry = 31;
+        private const int EventId_CrlCacheAcceptedFile = 32;
+        private const int EventId_CrlCacheWriteFailed = 33;
+        private const int EventId_CrlCacheWriteSucceeded = 34;
+        private const int EventId_CrlCheckOffline = 35;
+        private const int EventId_CrlChainFinished = 36;
+        private const int EventId_AllRevocationErrorsIgnored = 37;
+        private const int EventId_OcspResponseFromCache = 38;
+        private const int EventId_OcspResponseFromDownload = 39;
+        private const int EventId_RawElementStatus = 40;
+        private const int EventId_FinalElementStatus = 41;
+        private const int EventId_CouldNotOpenCAStore = 42;
+        private const int EventId_CachingIntermediate = 43;
+        private const int EventId_CachingIntermediateFailed = 44;
+        private const int EventId_RevocationCheckStart = 45;
+        private const int EventId_RevocationCheckStop = 46;
+        private const int EventId_CrlIdentifiersDetermined = 47;
+
+        private static string GetCertificateSubject(SafeX509Handle certHandle)
+        {
+            bool addedRef = false;
+
+            try
+            {
+                // Ensure that certHandle stays alive while we use an interior pointer.
+                certHandle.DangerousAddRef(ref addedRef);
+                X500DistinguishedName dn = Interop.Crypto.LoadX500Name(Interop.Crypto.X509GetSubjectName(certHandle));
+                return dn.Name;
+            }
+            finally
+            {
+                if (addedRef)
+                {
+                    certHandle.DangerousRelease();
+                }
+            }
+        }
+
+        internal bool ShouldLogElementStatuses()
+        {
+            return IsEnabled(EventLevel.Verbose, EventKeywords.None);
+        }
```

Comment:
@marek-safar, will this helper confuse the linker and its ability to remove EventSource-related code based on guards, or is this ok?